test/tests.js

```diff
@@ -0,0 +1,182 @@
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+
+let contract, owner, addr1, addr2, addr3
+
+describe('$LEVEL', function () {
+  beforeEach(async () => {
+    const Contract = await ethers.getContractFactory('Level')
+    contract = await Contract.deploy()
+    await contract.deployed()
+    const [o, a1, a2, a3] = await ethers.getSigners()
+    owner = o
+    addr1 = a1
+    addr2 = a2
+    addr3 = a3
+  })
+
+  describe('ERC-721 token', function () {
+    it('should be initialized with a name and symbol', async function () {
+      const name = await contract.name()
+      const symbol = await contract.symbol()
+      expect(name).to.equal('Level')
+      expect(symbol).to.equal('LEVEL')
+    })
+
+    describe('minting', function () {
+      it('should start at token id 0, and increment each mint', async function () {
+        await contract.connect(addr1).mint()
+        const owner1 = await contract.ownerOf(0)
+        expect(owner1).to.equal(addr1.address)
+        await contract.connect(addr2).mint()
+        const owner2 = await contract.ownerOf(1)
+        expect(owner2).to.equal(addr2.address)
+        await contract.connect(addr3).mint()
+        const owner3 = await contract.ownerOf(2)
+        expect(owner3).to.equal(addr3.address)
+      })
+      it('should only allow one token per address', async function () {
+        await contract.connect(addr1).mint()
+        await expect(contract.connect(addr1).mint()).to.be.revertedWith(
+          'Address can only have one LEVEL token!',
+        )
+      })
+      it('should only allow transfer during mint', async function () {
+        await contract.connect(addr1).mint()
+        await expect(
+          contract.connect(addr1).transferFrom(addr1.address, addr2.address, 0),
```

Comment:
I would like to see the "happy" version of this: when transferFrom would have a valid path. Just for my own understanding.

Reply:
Thanks for clarifying! @nicovalencia